testcontainers/src/watchdog.rs

```diff
@@ -0,0 +1,55 @@
+use lazy_static::lazy_static;
+use signal_hook::{
+    consts::{SIGINT, SIGQUIT, SIGTERM},
+    iterator::Signals,
+};
+use std::{collections::BTreeSet, sync::Mutex, thread};
+
+lazy_static! {
+    static ref WATCHDOG: Mutex<Watchdog> =
+        Mutex::new(Watchdog::new().expect("failed to register signal handler"));
+}
```

Comment:
Personally, I am more a fan of the macro-free version: https://docs.rs/conquer-once/latest/conquer_once/type.Lazy.html